x/distribution/keeper/calculation.go

```diff
@@ -0,0 +1,161 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+)
+
+// initialize rewards for a new validator
+func (k Keeper) initializeValidator(ctx sdk.Context, val sdk.Validator) {
+	// set initial historical rewards (period 0)
+	k.SetValidatorHistoricalRewards(ctx, val.GetOperator(), 0, types.ValidatorHistoricalRewards{})
+
+	// set current rewards (starting at period 1)
+	k.SetValidatorCurrentRewards(ctx, val.GetOperator(), types.ValidatorCurrentRewards{
+		Rewards: sdk.DecCoins{},
+		Period:  1,
+	})
+
+	// set accumulated commission
+	k.SetValidatorAccumulatedCommission(ctx, val.GetOperator(), types.ValidatorAccumulatedCommission{})
```

Comment:
maybe use of `InitialValidatorAccumulatedCommission` function would be good?

Reply:
Sure, done.